### PR TITLE
Self links contain embed params

### LINKED ISSUE #3062: Self links contain embed params
**Describe the bug**
https://github.com/DSpace/DSpace/pull/3030 has fixed most of the issues with self links that didn't contain all relevant parameters, but it seems to have introduced a new bug: embed parameters are now included in the self link. Since they don't affect the resource itself, only which _other_ resources are also included in the response, they should not be included.

**To Reproduce**
1. Retrieve any resource with an embed param. e.g. [this authorization ](https://dspace7.4science.cloud/server/#https://dspace7.4science.cloud/server/api/authz/authorizations/search/object?uri=https://dspace7.4science.cloud/server/api/core/sites/6d65c6a2-3fe7-44dd-bacb-79271257c35d&feature=epersonRegistration&embed=feature
)
2. Observe that, while the self link now contains all required params, it also contains the embed params

**Expected behavior**
The self link for a HAL resource should contain the parameters that have a (possible) effect on that resource but it shouldn't contain parameters that can't affect it.

**Related work**
This issue seems to have been introduced by https://github.com/DSpace/DSpace/pull/3030

### PR DESCRIPTION
**Pull request by jonas-atmire**
## References
* Fixes #3062
* Problem initially introduced by #3030 

## Description
Currently, self links should include embed parameters when returned, but this is not OK, since they don't affect the resource itself.
This PR avoids the addition of the aforementioned embed parameter
## Instructions for Reviewers

List of changes in this PR:
* RestResourceController -> When creating the self link (And including the actual parameters through the code from #3030), the embed parameter is excluded. The Projection creation is based on a [hardcoded check on the “embed” parameter](https://github.com/atmire/DSpace/blob/w2p-75404_self-links-contain-embeds/dspace-server-webapp/src/main/java/org/dspace/app/rest/utils/Utils.java#L542) as well, so this sort of check has been used similarly in this case
* RestResourceControllerIT.java -> A new test has been created that checks the steps mentioned below

**Test case**
* Retrieve any resource with an embed param. e.g. `/api/core/metadatafields/search/byFieldName?schema=dc&offset=0&embed=schema`
* The self link for a HAL resource should contain the parameters that have a (possible) effect on that resource but it shouldn't contain parameters that can't affect it.
    * /api/core/metadatafields/search/byFieldName?schema=dc&offset=0

## Checklist

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/RestResourceController.java [hunk 1]
```diff
@@ -1021,15 +1021,18 @@ public RepresentationModel listSearchMethods(@PathVariable String apiCategory, @
...
-     * @return encoded uriString containing request parameters
+     * @return encoded uriString containing request parameters without embed parameter
      */
     private String getEncodedParameterStringFromRequestParams(
             @RequestParam MultiValueMap<String, Object> parameters) {
         UriComponentsBuilder uriComponentsBuilder = UriComponentsBuilder.newInstance();
 
         for (String key : parameters.keySet()) {
-            uriComponentsBuilder.queryParam(key, parameters.get(key));
+            if (!StringUtils.equals(key, "embed")) {
```
Comment (tdonohue) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/RestResourceController.java:1033]: While I think this is correct based on current `main`, I think we're going to hit a "conflict" with this other open PR #3114 which adds an `embed.size` parameter.  I think we may need to consider loosening this `if` to make sure the param is not `embed` (exactly) or starts with `embed.`

If it makes it easier to test, you also could update your branch here with the changes in #3114... I'll leave that up to you though.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@jonas-atmire : Overall this code looks correct & the test looks correct.  However, as I noted below, I think this PR may overlap with #3114 ... so I think we may want to update your PR to ensure it's also filtering out `embed.*` params from self links.  I don't think we can strictly match on `embed` only...see more inline below.

Beyond that, I haven't tested this yet, but I see no other issues after an initial code review.

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/RestResourceController.java (hunk 1 above)
Comment (jonas-atmire) [reply dspace-server-webapp/src/main/java/org/dspace/app/rest/RestResourceController.java]: @tdonohue Good looking out for this other PR that also touched upon this part of the code.

We've subsequently merged in PR #3114, and updated the code/tests to handle this additional case of embed.xxx params

**Review by tdonohue — APPROVED** [review]
👍 Looks good now.  Thanks @jonas-atmire and @YanaDePauw  !

**Pull request merged**